lib/AST/CASTBridging.cpp

```diff
@@ -44,58 +44,116 @@ struct BridgedDiagnosticImpl {
 };
 } // namespace
 
+static inline BridgedDeclContext bridgedDeclContext(DeclContext *declContext) {
+  return {declContext};
+}
+
+static inline DeclContext *unbridged(BridgedDeclContext cDeclContext) {
+  return static_cast<DeclContext *>(cDeclContext.raw);
+}
+
+// Define `bridged` overloads for each AST node.
+#define AST_BRIDGING_WRAPPER(Name)                                             \
+  [[maybe_unused]]                                                             \
+  static Bridged##Name bridged(Name *raw) {                                    \
+    return {raw};                                                              \
+  }                                                                            \
+// Don't define `bridged` overloads for the TypeRepr subclasses, since we always
+// return a BridgedTypeRepr currently, only define `bridged(TypeRepr *)`.
+#define TYPEREPR(Id, Parent)
+#include "swift/AST/ASTBridgingWrappers.def"
+
+// Define `unbridged` overloads for each AST node.
+#define AST_BRIDGING_WRAPPER(Name)                                             \
+  [[maybe_unused]]                                                             \
+  static Name *unbridged(Bridged##Name bridged) {                              \
+    return static_cast<Name *>(bridged.raw);                                   \
+  }
+#include "swift/AST/ASTBridgingWrappers.def"
+
+// Define `.asDecl` on each BridgedXXXDecl type.
+#define DECL(Id, Parent)                                                       \
+  BridgedDecl Id##Decl_asDecl(Bridged##Id##Decl decl) {                        \
+    return bridged((Decl *)unbridged(decl));                                   \
```

Comment:
Any opinion on whether to use `static_cast` here? I think (not sure) that it does more checking and should be sufficient here. I don’t have really strong opinions here, though. So up to you.

Reply:
Heh yeah, I didn't really care since we know up-front that it's always an upcast, but yeah you're right `static_cast` is better